### PR TITLE
Added explanation as to why targets should be prefixed with package name

### PR DESCRIPTION
**Pull request by davetcoleman**
Based on discussion here:
https://github.com/catkin/catkin_tools/issues/236

### CODE REVIEW

File: src/catkin_pkg/templates/CMakeLists.txt.in [hunk 1]
```diff
@@ -121,6 +121,8 @@ catkin_package(
 # add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 
 ## Declare a C++ executable
+## Since catkin_make collects all cmake targets it is recommended to prefix all targets with
```
Comment (dirk-thomas) [inline src/catkin_pkg/templates/CMakeLists.txt.in]: `catkin_make collects all cmake targets` is a bit too fuzzy. I would propose the following text which explains more clearly what is happening here:

> With `catkin_make` all packages are built within a single CMake context
> The recommended prefix ensures that target names across packages don't collide

Comment (davetcoleman) [inline src/catkin_pkg/templates/CMakeLists.txt.in]: Thanks Dirk, updated and rebased

**Comment by dirk-thomas** [comment]
Thanks. LGTM.

**Comment by NikolausDemmel** [comment]
+1

**Comment by gavanderhoorn** [comment]
Might be nice to add that you can use `set_target_properties` like this:
```cmake
set_target_properties(${PROJECT_NAME}_node PROPERTIES OUTPUT_NAME node PREFIX "")
```
to remove the prefix from your binary names again. It's not needed as the package name will provide a scope when used with other ROS infrastructure.

**Comment by davetcoleman** [comment]
@gavanderhoorn that's a good point but I was thinking I didn't want to over-complicate the CMakeLists.txt even further. But thinking about it more, that does seem like a good idea.

**Comment by gavanderhoorn** [comment]
A valid point, but if we don't add the line, we'll have new users blindly copy/pasting this and end up with things like `rosrun someones_pkg someones_pkg_their_node`. Not very desirable I'd say.

**Comment by davetcoleman** [comment]
I've added documentation for this new feature

**Comment by dirk-thomas** [comment]
Thank you for iterating on this. I have changed the term `namespace` to `prefix`. The modified commit has been cherry-picked in 8574d5b9fabf3ba176ea21bc3629f15912e11fc1

**Comment by davetcoleman** [comment]
thanks!

**Pull request closed without merging**